src/App.tsx

```diff
@@ -15,36 +16,105 @@
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+interface SortParams {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortBy, isReversed }: SortParams) {
+  const preparedGoods = [...goods];
+
+  if (sortBy !== SortType.default) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType>(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortBy,
+    isReversed,
+  });
+
+  const handleResetButton = () => {
+    setSortBy(SortType.default);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortBy !== SortType.alphabet,
+          })}
+          onClick={() => setSortBy(SortType.alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortBy !== SortType.length,
+          })}
+          onClick={() => setSortBy(SortType.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
```

Comment:
[UX] - The 'Reverse' button should have an active state to show when the list is reversed. This visual feedback is crucial for a good user experience.